### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by soenkehahn**

### CODE REVIEW

File: src/System/Console/GetOpt/Generics/Modifier.hs [hunk 1]
```diff
@@ -3,12 +3,16 @@
...
+  mkModifiers,
   mkShortOptions,
-  mkLongOptions,
+  mkLongOption,
+
+  deriveShortOptions,
  ) where
 
 import           Data.List
+import qualified Data.Map.Strict                                as M
```
Comment (jkarni) [inline src/System/Console/GetOpt/Generics/Modifier.hs]: missing `containers` in the cabal file.

File: src/System/Console/GetOpt/Generics/Modifier.hs [hunk 2]
```diff
@@ -24,6 +28,32 @@ data Modifier
     --   through the @fieldName@ by @customName@.
   deriving (Show, Eq, Ord)
 
+data Modifiers = Modifiers {
+  _shortOptions :: M.Map String [Char],
```
Comment (jkarni) [inline src/System/Console/GetOpt/Generics/Modifier.hs]: Just out of curiosity - are you using `_` to indicate that it's not exported?

File: src/System/Console/GetOpt/Generics/Modifier.hs (hunk 1 above)
Comment (soenkehahn) [inline src/System/Console/GetOpt/Generics/Modifier.hs]: @sol: Would you remove this dependency? (Should be very easy, but it also comes with ghc.)

File: src/System/Console/GetOpt/Generics/Modifier.hs (hunk 2 above)
Comment (soenkehahn) [inline src/System/Console/GetOpt/Generics/Modifier.hs]: `ghc` complained about unused identifiers. But I still want to have them for documentation.

File: src/System/Console/GetOpt/Generics/Modifier.hs (hunk 1 above)
Comment (sol) [inline src/System/Console/GetOpt/Generics/Modifier.hs]: If it's easy to do so then yes, I would remove it

**Pull request merged**